### PR TITLE
feat: new transaction filter selectable accounts

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

The source account that has been selected by the user in the "new transaction" wizard should be filtered in the list of possible destination accounts.

# Changes

- add `filterIdentifier` to `SelectAccount` component
- use filter in "New Transaction" wizard

### CODE REVIEW

File: frontend/svelte/src/lib/components/accounts/SelectAccount.svelte [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
-      {#each $accountsStore.subAccounts as subAccount}
-        <AccountCard
-          role="button"
-          on:click={() => chooseAccount(subAccount)}
-          account={subAccount}>{subAccount.name}</AccountCard
-        >
-      {/each}
+    {#if filterIdentifier !== mainAccount.identifier}
+      {#if displayTitle}
+        <h4>{$i18n.accounts.my_accounts}</h4>
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/accounts/SelectAccount.svelte:32]: Why wouldn't you display the title if the you filter the Main account?

**Review by lmuntanerDfinity — COMMENTED** [review]
Just one question

File: frontend/svelte/src/lib/components/accounts/SelectAccount.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/accounts/SelectAccount.svelte:32]: But good point I can maybe remove `displayTitle` and make the `if` function of the `filterIdentifier`?

No filter => no display title
Filter => display title

Will do

Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/accounts/SelectAccount.svelte]: Done and also gotcha, title move before the checks. Now looks good.

File: frontend/svelte/src/lib/components/accounts/SelectAccount.svelte [hunk 2]
```diff
@@ -1,51 +1,54 @@
...
-    (accountStore) => {
-      mainAccount = accountStore?.main;
-    }
+  let subAccounts: Account[];
+  $: subAccounts = ($accountsStore?.subAccounts ?? []).filter(
+    ({ identifier }: Account) => identifier !== filterIdentifier
   );
 
-  onDestroy(unsubscribeAccounts);
+  // Display the "My Account" title only when we filter the list
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/accounts/SelectAccount.svelte]: Thanks. Long day, done.

**Pull request merged**